http-api-samples/http-api-owner-sample/src/main/java/org/fido/iot/api/OwnerSystemResourceServlet.java

```diff
@@ -0,0 +1,457 @@
+// Copyright 2020 Intel Corporation
+// SPDX-License-Identifier: Apache 2.0
+
+package org.fido.iot.api;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintWriter;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.Timestamp;
+import java.sql.Types;
+import java.util.Calendar;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.sql.DataSource;
+import org.fido.iot.protocol.Composite;
+import org.fido.iot.protocol.Const;
+import org.fido.iot.serviceinfo.ModuleManager;
+
+public class OwnerSystemResourceServlet extends HttpServlet {
+
+  public static final String BYTES_TAG = "bytes";
+  public static final String ID_TAG = "id";
+  public static final String CRID_TAG = "crid";
+  public static final String GUID_TAG = "guid";
+  public static final String DEVICE_TAG = "device";
+  public static final String OS_TAG = "os";
+  public static final String VERSION_TAG = "version";
+  public static final String ARCH_TAG = "arch";
+  public static final String FILENAME_TAG = "filename";
+  public static final String PRIORITY_TAG = "priority";
+  public static final String HASH_TAG = "hash";
+  public static final String MODULE_TAG = "module";
+  public static final String VAR_TAG = "var";
+  public static final String WHERE_AND = " AND ";
+
+
+  private byte[] getQueryBytes(String queryArg) {
+    StringBuilder hex = new StringBuilder();
+    for (char c : queryArg.toCharArray()) {
+      if (Character.isLetterOrDigit(c)) {
+        hex.append(c);
+      }
+    }
+
+    return Composite.decodeHex(hex.toString());
+  }
+
+  private String getWhereClause(HttpServletRequest req) {
+    String id = req.getParameter(ID_TAG);
+    String crid = req.getParameter(CRID_TAG);
+    String guid = req.getParameter(GUID_TAG);
+    String device = req.getParameter(DEVICE_TAG);
+    String os = req.getParameter(OS_TAG);
+    String version = req.getParameter(VERSION_TAG);
+    String arch = req.getParameter(ARCH_TAG);
+    String fileName = req.getParameter(FILENAME_TAG);
+    String priority = req.getParameter(PRIORITY_TAG);
+    String hash = req.getParameter(HASH_TAG);
+    String module = req.getParameter(MODULE_TAG);
+    String varName = req.getParameter(VAR_TAG);
+    String contentType = null;
+
+    if (module != null && varName != null) {
+      contentType = module + ModuleManager.MODULE_DELIMITER + varName;
+    }
+
+    StringBuilder whereString = new StringBuilder("WHERE ");
+
+    if (id != null) {
```

Comment:
Better practice would be to build a parameterized where clause (using ?) instead of adding each as a string to prevent sql injection (this would likely be identified in a security review).